### PR TITLE
Don't force rasterized output when exporting point pattern fills

### PR DESCRIPTION
**Pull request by nyalldawson**
This avoids the (very long standing!) issue of forced conversion to a raster based pattern which currently occurs when exporting maps/layouts to a vector format (e.g. PDF). The raster pattern
results in considerable quality loss, and the tiling edges of the raster brush can sometimes be seen in outputs.

Additionally, fixes render corrupt when marker subsymbols have data defined properties which affect the marker shape, such as data defined rotation or sizes

Refs #16100 (still needs fixing for line fill symbols)

### CODE REVIEW

**Comment by nyalldawson** [comment]
Example of the data defined raster brush tiling issue:

Using a random marker size with the fill pattern previously gave this:

image

Now you get the desired result:

image

Also note the dramatic quality improvement in the rendering quality - antialiasing is properly applied!

This fix allows creation of symbols which were previously corrupted, such as scree slopes:

image

image

image

**Comment by nyalldawson** [comment]
ping @andreasneumann

File: src/core/symbology/qgsfillsymbollayer.cpp [hunk 1]
```diff
@@ -3427,13 +3570,13 @@ void QgsPointPatternFillSymbolLayer::applyDataDefinedSettings( QgsSymbolRenderCo
   double offsetX = mOffsetX;
   if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyOffsetX ) )
   {
-    context.setOriginalValueVariable( mDisplacementX );
+    context.setOriginalValueVariable( mOffsetX );
```
Comment (nirvn) [inline src/core/symbology/qgsfillsymbollayer.cpp:3573]: Ouch -- thanks for fixing.

**Review by nirvn — APPROVED** [review]
Great job.

**Comment by nyalldawson** [comment]
Thanks @nirvn!

**Pull request merged**